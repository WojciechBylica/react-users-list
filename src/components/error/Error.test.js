import React from 'react';
import renderer from 'react-test-renderer';
import Error from '.';
import { shallow } from 'enzyme';

//jest snapshots:
describe('it renders ok', () => {
  it('should has default text if no prop provided', () => {
    const tree = renderer.create(<Error />);
    expect(tree).toMatchSnapshot();
  });
  it('should has text "testing error" if it was provided by props', () => {
    const tree = renderer.create(<Error message="testing error" />);
    expect(tree).toMatchSnapshot();
  });
});

//enzyme:
describe('it renders ok', () => {
  it('should has default text if no prop provided', () => {
    const error = shallow(<Error />);
    const paragraph = error.find('p');
    expect(paragraph.text()).toBe(
      'an error has occurred... please check your internet connection and try again...'
    );
  });
  it('should has text "error message" if provided from props', () => {
    const error = shallow(<Error message="error message" />);
    const paragraph = error.find('p');
    expect(paragraph.text()).toBe('error message');
  });
});
