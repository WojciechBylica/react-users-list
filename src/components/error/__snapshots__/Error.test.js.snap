// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders ok should has default text if no prop provided 1`] = `
<p
  className="error"
  data-test="error"
>
  an error has occurred... please check your internet connection and try again...
</p>
`;

exports[`it renders ok should has text "testing error" if it was provided by props 1`] = `
<p
  className="error"
  data-test="error"
>
  testing error
</p>
`;
